sdk/communication/Azure.Communication.PhoneNumbers/src/SipRouting/SipConfiguration.cs

```diff
@@ -0,0 +1,120 @@
+        {
+            writer.WriteStartObject();
+            if (Trunks != null)
+            {
+                writer.WritePropertyName("trunks");
+                writer.WriteStartObject();
+
+                foreach (KeyValuePair<string, SipTrunk> item in Trunks)
+                {
+                    if (item.Value != null)
+                    {
+                        writer.WritePropertyName(item.Key);
+                        writer.WriteObjectValue(item.Value);
+                    }
+                    else
+                    {
+                        writer.WriteNull(item.Key);
+                    }
+                }
+                writer.WriteEndObject();
+            }
+            if (Routes != null)
+            {
+                writer.WritePropertyName("routes");
+                writer.WriteStartArray();
+                foreach (SipTrunkRoute item in Routes)
+                {
+                    writer.WriteObjectValue(item);
+                }
+                writer.WriteEndArray();
+            }
+            writer.WriteEndObject();
+        }
+
+        internal static SipConfiguration DeserializeSipConfiguration(JsonElement element)
+        {
+            Optional<IReadOnlyDictionary<string, SipTrunk>> trunks = default;
+            Optional<IReadOnlyList<SipTrunkRoute>> routes = default;
+            foreach (var property in element.EnumerateObject())
+            {
+                if (property.NameEquals("trunks"))
+                {
+                    if (property.Value.ValueKind == JsonValueKind.Null)
+                    {
+                        property.ThrowNonNullablePropertyIsNull();
+                        continue;
+                    }
+                    Dictionary<string,SipTrunk> dictionary = new Dictionary<string, SipTrunk>();
+                    foreach (var property0 in property.Value.EnumerateObject())
+                    {
+                        if (property0.Value.ValueKind == JsonValueKind.Null)
+                        {
+                            dictionary.Add(property0.Name, null);
+                        }
+                        else
+                        {
+                            var sipTrunk = SipTrunk.DeserializeSipTrunk(property0.Value);
+                            sipTrunk.Fqdn = property0.Name;
+                            dictionary.Add(property0.Name, sipTrunk);
+                        }
+                    }
+                    trunks = dictionary;
+                    continue;
+                }
+                if (property.NameEquals("routes"))
+                {
+                    if (property.Value.ValueKind == JsonValueKind.Null)
+                    {
+                        property.ThrowNonNullablePropertyIsNull();
+                        continue;
+                    }
+                    List<SipTrunkRoute> array = new List<SipTrunkRoute>();
+                    foreach (var item in property.Value.EnumerateArray())
```

Comment:
nit: `property.Value.EnumerateArray().ForEach(item => array.Add(SipTrunkRoute.DeserializeSipTrunkRoute(item)));`

Reply:
Similar to previous comment, property.Value.EnumerateArray() returns an Enumerator, doesn't implement .ForEach().